### PR TITLE
CI failure FAQ

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
Add an FAQ-styled guide to some common kinds of CI failures. Also update the existing CI guide. In particular, add information on outcome analysis.

### CODE REVIEW

**Review by Harry-Ramsey — APPROVED** [review]
LGTM, useful documentation for new team joiners at ARM as well.

**Review by davidhorstmann-arm — APPROVED** [review]
LGTM, thanks!

File: kb/testing/understanding-ci-failures.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+## Understanding PR test failures
+
+### PR tests failed. Which components failed?
+
+If the PR tests fail, the status on GitHub is set to a string that lists failed components. Note that due to a GitHub limitation, the list of failed components may be truncated. Click the “Details” link to see the full job breakdown.
+
+The default view on Jenkins unfortunately truncates job names. For more convenient access to the names and logs of failed components:
+
+1. Install the GreaseMonkey/TamperMonkey user script [JenkinsPipelineSteps](https://openuserjs.org/scripts/magnushakansson/JenkinsPipelineSteps).
+2. Go to the Jenkins “classic” view by clicking the ![Go to classic](jenkins-blueocean-go-to-classic.svg) icon near the top right, then follow the “Pipeline Steps” link in the left column.
```
Comment (gilles-peskine-arm) [inline kb/testing/understanding-ci-failures.md:14]: I don't understand why the SVG icon isn't rendering in the document.

Comment (davidhorstmann-arm) [reply kb/testing/understanding-ci-failures.md:14]: [According to W3C](https://www.w3.org/Graphics/SVG/IG/resources/svgprimer.html#SVG_in_HTML):

> At the current time, of the five browsers, only Opera, Safari and Chrome support svg as an image format in HTML

Changing it to a PNG seems to work, I'll raise that as a PR.

Comment (davidhorstmann-arm) [reply kb/testing/understanding-ci-failures.md:14]: Hmm, though actually that document was a long time ago, so it's probably out of date.

Comment (tom-cosgrove-arm) [reply kb/testing/understanding-ci-failures.md:14]: https://caniuse.com/svg-img

**Pull request merged**